client/client.go

```diff
@@ -709,12 +708,90 @@ func (ec *SDKClient) GetGasPrice(
 			newGasPrice.Int(gasPrice)
 		}
 	} else {
-		log.Println("Setting existing gas price")
 		gasPrice = input.GasPrice
 	}
 	return gasPrice, nil
 }
 
+func (ec *SDKClient) GetGasTipCap(ctx context.Context, input Options) (*big.Int, error) {
+	if input.GasTipCap == nil {
+		var hex hexutil.Big
+		if err := ec.CallContext(ctx, &hex, "eth_maxPriorityFeePerGas"); err != nil {
+			return nil, err
+		}
+
+		return hex.ToInt(), nil
+	}
+
+	return input.GasTipCap, nil
+}
+
+func (ec *SDKClient) GetGasFeeCap(ctx context.Context, input Options, gasTipCap *big.Int) (*big.Int, error) {
+	if input.GasFeeCap == nil {
+		baseFee, err := ec.GetBaseFee(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		if baseFee != nil {
+			// Calculate max fee per gas
+			// Formula: GasFeeCap = max(BaseFeeMultiplier * BaseFee, BaseFeeFloor) + GasTipCap
+			baseFeeFloor := getBaseFeeFloor(ec.rosettaConfig)
```

Comment:
Why we specify a baseFeeFloor in rosetta config? 
Can we just make `gasFeeCap = gasTipCap + adjustedBaseFee`?

Reply:
@xiaying-peng will add comments based on our discussion

@ningcb yes, it's already configurable via env var, `ec.rosettaConfig.BaseFeeFloor` is an env var from downstream service (i.e. rosetta-base) and it's read in  `getBaseFeeFloor`